client/app/bundles/course/assessment/question/programming/actions/onlineEditorActionCreators.jsx

```diff
@@ -33,25 +33,81 @@ export function deletePythonTestCase(testType, index) {
   };
 }
 
-export function updateNewDataFile(filename, index) {
+export function updatePythonNewDataFile(filename, index) {
```

Comment:
yeah you're right, should actually refactor these into common functions as a base, and if any other programming languages requires specific functionalities for the actions, then additional language-specific actions should be defined on top of that (perhaps in a separate file).